client/src/app/employee/employee.component.ts

```diff
@@ -1,20 +1,41 @@
 import { Component } from '@angular/core';
-
-import { faLock, faUnlock } from '@fortawesome/free-solid-svg-icons';
+import { faPlus } from '@fortawesome/free-solid-svg-icons';
+import { Observable } from 'rxjs';
+import { Employee } from './employee-detail/state/employee.model';
+import { EmployeeQuery } from './employee-detail/state/employee.query';
+import { EmployeeService } from './employee-detail/state/employee.service';
+import { EmployeeListItem } from './employee-list/state/employee-list.model';
+import { EmployeeListQuery } from './employee-list/state/employee-list.query';
+import { EmployeeListService } from './employee-list/state/employee-list.service';
 
 @Component({
   selector: 'app-employee',
   templateUrl: './employee.component.html',
   styleUrls: ['./employee.component.scss'],
 })
 export class EmployeeComponent {
-  readonly faLock = faLock;
-  readonly faUnlock = faUnlock;
-  isReadonly = true;
+  readonly faPlus = faPlus;
+  employees: Observable<EmployeeListItem[]>;
+  employee: Observable<Employee>;
+  uiState = '';
+
+  constructor(
+    private employeeQuery: EmployeeQuery,
+    private employeeService: EmployeeService,
+    private employeeListService: EmployeeListService,
+    private employeeListQuery: EmployeeListQuery,
+  ) {
+    this.employee = this.employeeQuery.select((state) => state as Employee);
+    this.employees = this.employeeListQuery.selectAll();
+    this.employeeListService.getEmployees();
+  }
 
-  constructor() {}
+  setCreate() {
+    this.uiState = 'create';
+  }
 
-  toggleReadonly() {
-    this.isReadonly = !this.isReadonly;
+  setActive(employee: EmployeeListItem) {
```

Comment:
The Angular style guide recommends naming event handling methods (either from user events like `(click)` or custom component events) with an `on` prefix.

So this method would be named `onSelectEmployee()` and `setCreate()` would become `onCreate()`.

This is helpful from a tooling perspective, beyond just having a convention for naming, because component methods used in the template must be `public` (in Typescript, no access modifier implies `public`), but the Typescript language service doesn't know that the method is referenced in the template so it will display the number of references of the method as 0.

However, by giving the method the `on` prefix, you know that it's used in the template.

Other methods, if they have 0 references, you can safely remove because you know they aren't being used.

Reply:
Good to know, i like this convention. I need to read the Angular style guide more.